### PR TITLE
Update RoBERTa vocabulary files

### PR DESCRIPTION
**Pull request by gpengzhi**
```python
import torch
roberta = torch.hub.load('pytorch/fairseq', 'roberta.base')
roberta.eval()

tokens = roberta.encode('Hello world!')
print(tokens)  # [    0, 31414,   232,   328,     2]
```

```python
import texar.torch as tx
tokenizer = tx.data.RoBERTaTokenizer(pretrained_model_name='roberta-base')

input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)
print(input_ids)  # [0, 31414, 232, 328, 2]
```

### CODE REVIEW

**Comment by gpengzhi** [comment]
resolve #246

File: texar/torch/data/tokenizers/gpt2_tokenizer.py [hunk 1]
```diff
@@ -70,6 +70,24 @@ class GPT2Tokenizer(TokenizerBase, PretrainedGPT2Mixin):
...
+    _VOCAB_FILE_MAP = {
+        'vocab_file': {
+            'gpt2-small': 'encoder.json',
+            'gpt2-medium': 'encoder.json',
+            'gpt2-large': 'encoder.json',
+            'gpt2-xl': 'encoder.json',
+            '117M': 'encoder.json',
+            '345M': 'encoder.json',
+        },
+        'merges_file': {
```
Comment (ZhitingHu) [inline texar/torch/data/tokenizers/gpt2_tokenizer.py]: what does `merges` mean here?

File: texar/torch/data/tokenizers/roberta_tokenizer.py [hunk 2]
```diff
@@ -24,8 +24,7 @@
     'RoBERTaTokenizer',
 ]
 
-_GPT2_PATH = "https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/"
-_CHECKPOINT_FILES = ["encoder.json", "vocab.bpe"]
+_ROBERTA_PATH = "https://s3.amazonaws.com/models.huggingface.co/bert/"
```
Comment (ZhitingHu) [inline texar/torch/data/tokenizers/roberta_tokenizer.py]: Can we host the checkpoints by ourselves, on the asyml google drive?

File: texar/torch/data/tokenizers/gpt2_tokenizer.py (hunk 1 above)
Comment (gpengzhi) [reply texar/torch/data/tokenizers/gpt2_tokenizer.py]: BPE generates subword tokens by merging the most frequent consecutive byte pairs. This file contains all the merged byte pairs.

File: texar/torch/data/tokenizers/roberta_tokenizer.py (hunk 2 above)
Comment (gpengzhi) [reply texar/torch/data/tokenizers/roberta_tokenizer.py]: We can. Do we currently have asyml google drive?

**Comment by gpengzhi** [comment]
> `input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)`
> 
> is `max_seq_length` necessary here? what's the result in this case without setting `max_seq_length`?

It will be

**Comment by gpengzhi** [comment]
> `input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)`
> 
> is `max_seq_length` necessary here? what's the result in this case without setting `max_seq_length`?

`max_seq_length` is used here just to show the same result between ours and theirs. If `max_seq_length` is not specified, it will be `[0, 31414, 232, 328, 2, 0, 0, 0, ..., 0]`(zero-pad up to the maximum sequence length).

File: texar/torch/data/tokenizers/roberta_tokenizer.py (hunk 2 above)
Comment (ZhitingHu) [reply texar/torch/data/tokenizers/roberta_tokenizer.py]: > > `input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)`
> > is `max_seq_length` necessary here? what's the result in this case without setting `max_seq_length`?
> 
> `max_seq_length` is used here just to show the same result between ours and theirs. If `max_seq_length` is not specified, it will be `[0, 31414, 232, 328, 2, 0, 0, 0, ..., 0]`(zero-pad up to the maximum sequence length).

If the user doesn't want padding, it's difficult for them to know the correct seq length just as in this case ('Hello world!' needs seq_length 5).. Can we add an argument (or allow a special value of `max_seq_length`) so that the user can get the encoded text without padding (and without need to specify `max-seq-length` explicitly)?

**Comment by ZhitingHu** [comment]
> > `input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)`
> > is `max_seq_length` necessary here? what's the result in this case without setting `max_seq_length`?
> 
> `max_seq_length` is used here just to show the same result between ours and theirs. If `max_seq_length` is not specified, it will be `[0, 31414, 232, 328, 2, 0, 0, 0, ..., 0]`(zero-pad up to the maximum sequence length).

If the user doesn't want padding, it's difficult for them to know the correct seq length just as in this case ('Hello world!' needs seq_length 5).. Can we add an argument (or allow a special value of max_seq_length) so that the user can get the encoded text without padding (and without need to specify max-seq-length explicitly)?

**Comment by gpengzhi** [comment]
> > > `input_ids, _ = tokenizer.encode_text('Hello world!', max_seq_length=5)`
> > > is `max_seq_length` necessary here? what's the result in this case without setting `max_seq_length`?
> > 
> > 
> > `max_seq_length` is used here just to show the same result between ours and theirs. If `max_seq_length` is not specified, it will be `[0, 31414, 232, 328, 2, 0, 0, 0, ..., 0]`(zero-pad up to the maximum sequence length).
> 
> If the user doesn't want padding, it's difficult for them to know the correct seq length just as in this case ('Hello world!' needs seq_length 5).. Can we add an argument (or allow a special value of max_seq_length) so that the user can get the encoded text without padding (and without need to specify max-seq-length explicitly)?

There are two returns, `input_ids` and `input_mask`, for the function `encode_text`. If the user can compute the correct sequence length by aceessing `input_mask`. In this case, `input_mask` is `[1, 1, 1, 1, 1, 0, 0, ..., 0]`.

**Comment by ZhitingHu** [comment]
pls try to pass the test and merge asap

**Comment by gpengzhi** [comment]
I tried to disable `codecov/patch` because some of our tests (pre-trained tests) are done locally. `codecov/patch` is also reported to be not very reliable. Some other projects have met such issues (https://github.com/argoproj/argo-cd/pull/1926).  I think `codecov/project` is good enough to check the code quality for now. I tried many ways to disable `codecov/patch` but failed. It seems that someone also met a similar problem (https://community.codecov.io/t/cannot-disable-codecov-patch-check/682/4). Do you think it is reasonable to merge PRs regardless of the status of `codecov/patch`? @ZhitingHu

**Comment by ZhitingHu** [comment]
As long as the build is "passing" we're good for now

**Comment by gpengzhi** [comment]
https://github.com/pytorch/fairseq/issues/1432
